src/main/java/Parser.java

```diff
@@ -0,0 +1,33 @@
+package duke;
+
+import java.util.HashMap;
+
+public class Parser {
+    private static HashMap<String, CommandBuilder> registeredCommands = new HashMap<>();
```

Comment:
Consider initializing 'registeredCommands' using the 'Map<>' interface instead of 'HashMap<>' for improved flexibility and adherence to coding to interfaces best practices. So something like Map<String, CommandBuilder> registeredCommands = new HashMap<>()